agent/api/task.go

```diff
@@ -409,12 +417,30 @@ func (task *Task) dockerHostConfig(container *Container, dockerContainerMap map[
 		return nil, &HostConfigError{err.Error()}
 	}
 
+	// Populate hostConfig
 	hostConfig := &docker.HostConfig{
 		Links:        dockerLinkArr,
 		Binds:        binds,
 		PortBindings: dockerPortMap,
 		VolumesFrom:  volumesFrom,
 	}
+	// Update cgroup parent
+	// TODO: Feature gating
+	if task.CgroupEnabled() {
```

Comment:
As per @samuelkarp  comment, we definitely need to move this block to `task_unix.go`. We should not evaluate `if task.CgroupEnabled()` if running on windows.